### ISSUE TITLE
Compilation fail: SDL related...

### ISSUE THREAD

**Issue by fjtrujy** [ps2dev/ps2sdk-ports#32]
Hello, 
I have done a clean installation in a VM with Ubuntu 16 and is not compiling.
So far it looks that is not able to find SDL.h, so strange, few commits ago is still working.

This is the last part of the compilation log.

```
make -C sdl
make[1]: Entering directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl'
mkdir -p lib
make -C src
make[2]: Entering directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl/src'
GSKIT not set and gsKit not installed
make[2]: Leaving directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl/src'
make[1]: Leaving directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl'
make -C sdl install
make[1]: Entering directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl'
mkdir -p lib
make -C src
make[2]: Entering directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl/src'
GSKIT not set and gsKit not installed
make[2]: Leaving directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl/src'
make -C src install
make[2]: Entering directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl/src'
GSKIT not set and gsKit not installed
make[2]: Leaving directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl/src'
make[1]: Leaving directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl'
make -C sdl clean
make[1]: Entering directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl'
make -C src clean
make[2]: Entering directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl/src'
rm -f ../lib/libsdl.a ../lib/libsdlmain.a SDL.o SDL_error.o SDL_fatal.o SDL_getenv.o SDL_loadso.o video/SDL_video.o video/SDL_surface.o video/SDL_pixels.o video/SDL_blit.o video/SDL_blit_0.o video/SDL_blit_1.o video/SDL_blit_N.o video/SDL_blit_A.o  video/SDL_RLEaccel.o video/SDL_cursor.o video/SDL_bmp.o video/SDL_yuv.o video/SDL_yuv_sw.o video/SDL_stretch.o thread/SDL_thread.o  events/SDL_events.o events/SDL_quit.o events/SDL_active.o events/SDL_keyboard.o events/SDL_mouse.o events/SDL_resize.o thread/generic/SDL_syscond.o thread/generic/SDL_sysmutex.o audio/SDL_wave.o audio/SDL_audio.o audio/SDL_audiomem.o audio/SDL_audiocvt.o audio/SDL_mixer.o file/SDL_rwops.o cdrom/SDL_cdrom.o  timer/SDL_timer.o  joystick/SDL_joystick.o  endian/SDL_endian.o main/ps2sdk/sdl_main.o thread/ps2sdk/SDL_syssem.o thread/ps2sdk/SDL_systhread.o timer/ps2sdk/SDL_systimer.o video/ps2sdk/SDL_ps2video.o audio/ps2sdk/SDL_ps2audio.o  cdrom/ps2sdk/SDL_syscdrom.o joystick/ps2sdk/ps2joystick.o audio/ps2sdk/audsrv_irx.o  audio/ps2sdk/freesd_irx.o joystick/ps2sdk/sio2man_irx.o joystick/ps2sdk/padman_irx.o joystick/ps2sdk/mtapman_irx.o main/arch.o main/ps2sdk/sdl_main.o audio/ps2sdk/freesd_irx.c joystick/ps2sdk/sio2man_irx.c joystick/ps2sdk/padman_irx.c joystick/ps2sdk/mtapman_irx.c audio/ps2sdk/audsrv_irx.c video/ps2sdk/ps2kbd_irx.c video/ps2sdk/ps2mouse_irx.c video/ps2sdk/ps2usbd_irx.c
make[2]: Leaving directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl/src'
make -C test clean
make[2]: Entering directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl/test'
rm -f testtimer.elf testcdrom.elf testendian.elf testerror.elf testjoystick.elf testver.elf checkkeys.elf testtimer.o testcdrom.o testendian.o testerror.o testjoystick.o testver.o checkkeys.o
make[2]: Leaving directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl/test'
rm -f -r lib
make[1]: Leaving directory '/home/fjtrujy/Projects/ps2sdk-ports/sdl'
make -C sdlimage
make[1]: Entering directory '/home/fjtrujy/Projects/ps2sdk-ports/sdlimage'
ee-gcc -D_EE -O2 -G0 -Wall -DLOAD_BMP -DLOAD_GIF -DLOAD_JPG -DLOAD_LBM -DLOAD_PCX -DLOAD_PNG -DLOAD_PNM -DLOAD_TGA -DLOAD_XCF -DLOAD_XPM -DLOAD_TIF -I/usr/local/ps2dev/ps2sdk/ee/include -I/usr/local/ps2dev/ps2sdk/common/include -I. -I./include -I/usr/local/ps2dev/ps2sdk/ports/include -I/usr/local/ps2dev/ps2sdk/ports/include/zlib -I/usr/local/ps2dev/ps2sdk/ports/include/SDL -c IMG.c -o IMG.o
In file included from IMG.c:31:
include/SDL_image.h:30:17: SDL.h: No such file or directory
include/SDL_image.h:31:25: SDL_version.h: No such file or directory
include/SDL_image.h:32:24: begin_code.h: No such file or directory
In file included from IMG.c:31:
include/SDL_image.h:59: syntax error before "const"
include/SDL_image.h:59: warning: type defaults to `int' in declaration of `IMG_Linked_Version'
include/SDL_image.h:59: warning: data definition has no type or storage class
include/SDL_image.h:69: syntax error before "SDL_Surface"
include/SDL_image.h:69: syntax error before '*' token
include/SDL_image.h:69: warning: type defaults to `int' in declaration of `IMG_LoadTyped_RW'
include/SDL_image.h:69: warning: data definition has no type or storage class
include/SDL_image.h:71: syntax error before "SDL_Surface"
include/SDL_image.h:71: warning: type defaults to `int' in declaration of `IMG_Load'
include/SDL_image.h:71: warning: data definition has no type or storage class
include/SDL_image.h:72: syntax error before "SDL_Surface"
include/SDL_image.h:72: syntax error before '*' token
include/SDL_image.h:72: warning: type defaults to `int' in declaration of `IMG_Load_RW'
include/SDL_image.h:72: warning: data definition has no type or storage class
include/SDL_image.h:77: syntax error before "int"
include/SDL_image.h:77: warning: type defaults to `int' in declaration of `IMG_InvertAlpha'
include/SDL_image.h:77: warning: data definition has no type or storage class
include/SDL_image.h:80: syntax error before "int"
include/SDL_image.h:80: syntax error before '*' token
include/SDL_image.h:80: warning: type defaults to `int' in declaration of `IMG_isBMP'
include/SDL_image.h:80: warning: data definition has no type or storage class
include/SDL_image.h:81: syntax error before "int"
include/SDL_image.h:81: syntax error before '*' token
include/SDL_image.h:81: warning: type defaults to `int' in declaration of `IMG_isPNM'
include/SDL_image.h:81: warning: data definition has no type or storage class
include/SDL_image.h:82: syntax error before "int"
include/SDL_image.h:82: syntax error before '*' token
include/SDL_image.h:82: warning: type defaults to `int' in declaration of `IMG_isXPM'
include/SDL_image.h:82: warning: data definition has no type or storage class
include/SDL_image.h:83: syntax error before "int"
include/SDL_image.h:83: syntax error before '*' token
include/SDL_image.h:83: warning: type defaults to `int' in declaration of `IMG_isXCF'
include/SDL_image.h:83: warning: data definition has no type or storage class
include/SDL_image.h:84: syntax error before "int"
include/SDL_image.h:84: syntax error before '*' token
include/SDL_image.h:84: warning: type defaults to `int' in declaration of `IMG_isPCX'
include/SDL_image.h:84: warning: data definition has no type or storage class
include/SDL_image.h:85: syntax error before "int"
include/SDL_image.h:85: syntax error before '*' token
include/SDL_image.h:85: warning: type defaults to `int' in declaration of `IMG_isGIF'
include/SDL_image.h:85: warning: data definition has no type or storage class
include/SDL_image.h:86: syntax error before "int"
include/SDL_image.h:86: syntax error before '*' token
include/SDL_image.h:86: warning: type defaults to `int' in declaration of `IMG_isJPG'
include/SDL_image.h:86: warning: data definition has no type or storage class
include/SDL_image.h:87: syntax error before "int"
include/SDL_image.h:87: syntax error before '*' token
include/SDL_image.h:87: warning: type defaults to `int' in declaration of `IMG_isTIF'
include/SDL_image.h:87: warning: data definition has no type or storage class
include/SDL_image.h:88: syntax error before "int"
include/SDL_image.h:88: syntax error before '*' token
include/SDL_image.h:88: warning: type defaults to `int' in declaration of `IMG_isPNG'
include/SDL_image.h:88: warning: data definition has no type or storage class
include/SDL_image.h:89: syntax error before "int"
include/SDL_image.h:89: syntax error before '*' token
include/SDL_image.h:89: warning: type defaults to `int' in declaration of `IMG_isLBM'
include/SDL_image.h:89: warning: data definition has no type or storage class
include/SDL_image.h:92: syntax error before "SDL_Surface"
include/SDL_image.h:92: syntax error before '*' token
include/SDL_image.h:92: warning: type defaults to `int' in declaration of `IMG_LoadBMP_RW'
include/SDL_image.h:92: warning: data definition has no type or storage class
include/SDL_image.h:93: syntax error before "SDL_Surface"
include/SDL_image.h:93: syntax error before '*' token
include/SDL_image.h:93: warning: type defaults to `int' in declaration of `IMG_LoadPNM_RW'
include/SDL_image.h:93: warning: data definition has no type or storage class
include/SDL_image.h:94: syntax error before "SDL_Surface"
include/SDL_image.h:94: syntax error before '*' token
include/SDL_image.h:94: warning: type defaults to `int' in declaration of `IMG_LoadXPM_RW'
include/SDL_image.h:94: warning: data definition has no type or storage class
include/SDL_image.h:95: syntax error before "SDL_Surface"
include/SDL_image.h:95: syntax error before '*' token
include/SDL_image.h:95: warning: type defaults to `int' in declaration of `IMG_LoadXCF_RW'
include/SDL_image.h:95: warning: data definition has no type or storage class
include/SDL_image.h:96: syntax error before "SDL_Surface"
include/SDL_image.h:96: syntax error before '*' token
include/SDL_image.h:96: warning: type defaults to `int' in declaration of `IMG_LoadPCX_RW'
include/SDL_image.h:96: warning: data definition has no type or storage class
include/SDL_image.h:97: syntax error before "SDL_Surface"
include/SDL_image.h:97: syntax error before '*' token
include/SDL_image.h:97: warning: type defaults to `int' in declaration of `IMG_LoadGIF_RW'
include/SDL_image.h:97: warning: data definition has no type or storage class
include/SDL_image.h:98: syntax error before "SDL_Surface"
include/SDL_image.h:98: syntax error before '*' token
include/SDL_image.h:98: warning: type defaults to `int' in declaration of `IMG_LoadJPG_RW'
include/SDL_image.h:98: warning: data definition has no type or storage class
include/SDL_image.h:99: syntax error before "SDL_Surface"
include/SDL_image.h:99: syntax error before '*' token
include/SDL_image.h:99: warning: type defaults to `int' in declaration of `IMG_LoadTIF_RW'
include/SDL_image.h:99: warning: data definition has no type or storage class
include/SDL_image.h:100: syntax error before "SDL_Surface"
include/SDL_image.h:100: syntax error before '*' token
include/SDL_image.h:100: warning: type defaults to `int' in declaration of `IMG_LoadPNG_RW'
include/SDL_image.h:100: warning: data definition has no type or storage class
include/SDL_image.h:101: syntax error before "SDL_Surface"
include/SDL_image.h:101: syntax error before '*' token
include/SDL_image.h:101: warning: type defaults to `int' in declaration of `IMG_LoadTGA_RW'
include/SDL_image.h:101: warning: data definition has no type or storage class
include/SDL_image.h:102: syntax error before "SDL_Surface"
include/SDL_image.h:102: syntax error before '*' token
include/SDL_image.h:102: warning: type defaults to `int' in declaration of `IMG_LoadLBM_RW'
include/SDL_image.h:102: warning: data definition has no type or storage class
include/SDL_image.h:104: syntax error before "SDL_Surface"
include/SDL_image.h:104: warning: type defaults to `int' in declaration of `IMG_ReadXPMFromArray'
include/SDL_image.h:104: warning: data definition has no type or storage class
include/SDL_image.h:114:24: close_code.h: No such file or directory
IMG.c:38: syntax error before '*' token
IMG.c:39: warning: no semicolon at end of struct or union
IMG.c:40: warning: type defaults to `int' in declaration of `supported'
IMG.c:42: warning: braces around scalar initializer
IMG.c:42: warning: (near initialization for `supported[0]')
IMG.c:42: warning: initialization makes integer from pointer without a cast
IMG.c:42: warning: excess elements in scalar initializer
IMG.c:42: warning: (near initialization for `supported[0]')
IMG.c:42: warning: excess elements in scalar initializer
IMG.c:42: warning: (near initialization for `supported[0]')
IMG.c:43: warning: braces around scalar initializer
IMG.c:43: warning: (near initialization for `supported[1]')
IMG.c:43: warning: initialization makes integer from pointer without a cast
IMG.c:43: warning: excess elements in scalar initializer
IMG.c:43: warning: (near initialization for `supported[1]')
IMG.c:43: warning: excess elements in scalar initializer
IMG.c:43: warning: (near initialization for `supported[1]')
IMG.c:44: warning: braces around scalar initializer
IMG.c:44: warning: (near initialization for `supported[2]')
IMG.c:44: warning: initialization makes integer from pointer without a cast
IMG.c:44: warning: excess elements in scalar initializer
IMG.c:44: warning: (near initialization for `supported[2]')
IMG.c:44: warning: excess elements in scalar initializer
IMG.c:44: warning: (near initialization for `supported[2]')
IMG.c:45: warning: braces around scalar initializer
IMG.c:45: warning: (near initialization for `supported[3]')
IMG.c:45: warning: initialization makes integer from pointer without a cast
IMG.c:45: warning: excess elements in scalar initializer
IMG.c:45: warning: (near initialization for `supported[3]')
IMG.c:45: warning: excess elements in scalar initializer
IMG.c:45: warning: (near initialization for `supported[3]')
IMG.c:46: warning: braces around scalar initializer
IMG.c:46: warning: (near initialization for `supported[4]')
IMG.c:46: warning: initialization makes integer from pointer without a cast
IMG.c:46: warning: excess elements in scalar initializer
IMG.c:46: warning: (near initialization for `supported[4]')
IMG.c:46: warning: excess elements in scalar initializer
IMG.c:46: warning: (near initialization for `supported[4]')
IMG.c:47: warning: braces around scalar initializer
IMG.c:47: warning: (near initialization for `supported[5]')
IMG.c:47: warning: initialization makes integer from pointer without a cast
IMG.c:47: warning: excess elements in scalar initializer
IMG.c:47: warning: (near initialization for `supported[5]')
IMG.c:47: warning: excess elements in scalar initializer
IMG.c:47: warning: (near initialization for `supported[5]')
IMG.c:48: warning: braces around scalar initializer
IMG.c:48: warning: (near initialization for `supported[6]')
IMG.c:48: warning: initialization makes integer from pointer without a cast
IMG.c:48: warning: excess elements in scalar initializer
IMG.c:48: warning: (near initialization for `supported[6]')
IMG.c:48: warning: excess elements in scalar initializer
IMG.c:48: warning: (near initialization for `supported[6]')
IMG.c:49: warning: braces around scalar initializer
IMG.c:49: warning: (near initialization for `supported[7]')
IMG.c:49: warning: initialization makes integer from pointer without a cast
IMG.c:49: warning: excess elements in scalar initializer
IMG.c:49: warning: (near initialization for `supported[7]')
IMG.c:49: warning: excess elements in scalar initializer
IMG.c:49: warning: (near initialization for `supported[7]')
IMG.c:50: warning: braces around scalar initializer
IMG.c:50: warning: (near initialization for `supported[8]')
IMG.c:50: warning: initialization makes integer from pointer without a cast
IMG.c:50: warning: excess elements in scalar initializer
IMG.c:50: warning: (near initialization for `supported[8]')
IMG.c:50: warning: excess elements in scalar initializer
IMG.c:50: warning: (near initialization for `supported[8]')
IMG.c:51: warning: braces around scalar initializer
IMG.c:51: warning: (near initialization for `supported[9]')
IMG.c:51: warning: initialization makes integer from pointer without a cast
IMG.c:51: warning: excess elements in scalar initializer
IMG.c:51: warning: (near initialization for `supported[9]')
IMG.c:51: warning: excess elements in scalar initializer
IMG.c:51: warning: (near initialization for `supported[9]')
IMG.c:52: warning: braces around scalar initializer
IMG.c:52: warning: (near initialization for `supported[10]')
IMG.c:52: warning: initialization makes integer from pointer without a cast
IMG.c:52: warning: excess elements in scalar initializer
IMG.c:52: warning: (near initialization for `supported[10]')
IMG.c:52: warning: excess elements in scalar initializer
IMG.c:52: warning: (near initialization for `supported[10]')
IMG.c:53: warning: data definition has no type or storage class
IMG.c:55: syntax error before '*' token
IMG.c:56: warning: return type defaults to `int'
IMG.c:56: conflicting types for `IMG_Linked_Version'
include/SDL_image.h:59: previous declaration of `IMG_Linked_Version'
IMG.c: In function `IMG_Linked_Version':
IMG.c:57: syntax error before "linked_version"
IMG.c:58: `linked_version' undeclared (first use in this function)
IMG.c:58: (Each undeclared identifier is reported only once
IMG.c:58: for each function it appears in.)
IMG.c: At top level:
IMG.c:63: syntax error before '*' token
IMG.c:64: warning: return type defaults to `int'
IMG.c:64: conflicting types for `IMG_Load'
include/SDL_image.h:71: previous declaration of `IMG_Load'
IMG.c: In function `IMG_Load':
IMG.c:65: `SDL_RWops' undeclared (first use in this function)
IMG.c:65: `src' undeclared (first use in this function)
IMG.c:65: warning: implicit declaration of function `SDL_RWFromFile'
IMG.c:74: warning: return makes pointer from integer without a cast
IMG.c: At top level:
IMG.c:78: syntax error before '*' token
IMG.c:78: syntax error before '*' token
IMG.c:79: warning: return type defaults to `int'
IMG.c:79: conflicting types for `IMG_Load_RW'
include/SDL_image.h:72: previous declaration of `IMG_Load_RW'
IMG.c: In function `IMG_Load_RW':
IMG.c:80: `src' undeclared (first use in this function)
IMG.c:80: `freesrc' undeclared (first use in this function)
IMG.c:80: warning: return makes pointer from integer without a cast
IMG.c: At top level:
IMG.c:97: syntax error before '*' token
IMG.c:97: syntax error before '*' token
IMG.c:98: warning: return type defaults to `int'
IMG.c:98: conflicting types for `IMG_LoadTyped_RW'
include/SDL_image.h:69: previous declaration of `IMG_LoadTyped_RW'
IMG.c: In function `IMG_LoadTyped_RW':
IMG.c:100: `SDL_Surface' undeclared (first use in this function)
IMG.c:100: `image' undeclared (first use in this function)
IMG.c:103: `src' undeclared (first use in this function)
IMG.c:104: warning: implicit declaration of function `SDL_SetError'
IMG.c:109: warning: implicit declaration of function `SDL_RWseek'
IMG.c:111: `freesrc' undeclared (first use in this function)
IMG.c:112: warning: implicit declaration of function `SDL_RWclose'
IMG.c:117: warning: implicit declaration of function `SDL_RWtell'
IMG.c:120: request for member `is' in something not a structure or union
IMG.c:122: request for member `is' in something not a structure or union
IMG.c:127: `type' undeclared (first use in this function)
IMG.c:127: request for member `type' in something not a structure or union
IMG.c:135: request for member `load' in something not a structure or union
../Makefile.eeglobal:46: recipe for target 'IMG.o' failed
make[1]: *** [IMG.o] Error 1
make[1]: Leaving directory '/home/fjtrujy/Projects/ps2sdk-ports/sdlimage'
Makefile:101: recipe for target 'sdlimage' failed
make: *** [sdlimage] Error 2
```

Here you have attached the whole log.
[compilation.log](https://github.com/ps2dev/ps2sdk-ports/files/2726316/compilation.log)

Thanks

**Comment by sp193** [comment]
I see "GSKIT not set and gsKit not installed". Maybe you don't have GSKIT installed?
As of recently, you must also have the GSKIT global variable set as well.

**Comment by fjtrujy** [comment]
I have installed previously the GSKit, but obviously I didn't define any GSKit on the variables.
So, now is mandatory to have also a GSKit Variable? 

After read again all the Readmes (toolchain, GSKit, ps2sdk-ports) I don't see where you said that the GSKit variable is needed to be defined.

I'm right?

**Comment by sp193** [comment]
Yes. So why not just add it in? Nobody updated the README, that's all.
The use of the GSKIT environmental variable was not mandated by the gsKit developer(s). I'm now just a custodian of this repository, but I am not the solo developer of these projects.

There is a need to determine whether gsKit exists. Some projects assume gsKit is installed under $GSKIT/gsKit, while some others assume it is installed under $PS2DEV/gsKit
This ambiguity has been around for a long time and nothing was done, until recently. As you can see in #4, the $GSKIT environmental variable won. For better or worse, as we have to eventually choose one way.

**Comment by fjtrujy** [comment]
Here you have the PR https://github.com/ps2dev/gsKit/pull/25.

Anyway, this issue will disappear as soon as we decide to merge the `gsKit` into the `PS2SDK`.
In the readme you mention that the idea is to be merged in the future, do you think that this future is going to be soon? or there is still a long road pending to implement?

Thanks.

**Comment by sp193** [comment]
I did not write that README.

But how would merging it into the PS2SDK solve all this? It's caused by some people actually using the GSKIT variable, which led to their projects depending on it.

**Comment by fjtrujy** [comment]
Yeah, obviously merging into `PS2SDK` won't fix the issue automatically, the other projects need to be updated.
But if you think that way, you will always have the `GSKit` if you have the `PS2SDK`....
But well this is something for the future....
Thanks!
